services/api-server/src/simcore_service_api_server/api/errors/httpx_client_error.py

```diff
@@ -1,50 +1,67 @@
 """ General handling of httpx-based exceptions
 
     - httpx-based clients are used to communicate with other backend services
-    - When those respond with 4XX, 5XX status codes, those are generally handled here
+    - any exception raised by a httpx client will be handled here.
 """
 import logging
+from typing import Any
 
-from fastapi import status
-from httpx import HTTPStatusError
-from starlette.requests import Request
-from starlette.responses import JSONResponse
-
-from .http_error import create_error_json_response
+from fastapi import HTTPException, Request, status
+from httpx import HTTPError, HTTPStatusError, TimeoutException
 
 _logger = logging.getLogger(__file__)
 
 
-async def httpx_client_error_handler(_: Request, exc: HTTPStatusError) -> JSONResponse:
+async def handle_httpx_client_exceptions(_: Request, exc: HTTPError):
+    """
+    Default httpx exception handler.
+    See https://www.python-httpx.org/exceptions/
+    With this in place only HTTPStatusErrors need to be customized closer to the httpx client itself.
     """
-    This is called when HTTPStatusError was raised and reached the outermost handler
+    status_code: Any
+    detail: str
+    headers: dict[str, str] = {}
+    if isinstance(exc, HTTPStatusError):
+        status_code, detail, headers = _handle_httpx_status_exceptions(exc)
+    elif isinstance(exc, TimeoutException):
+        status_code = status.HTTP_504_GATEWAY_TIMEOUT
+        detail = f"Request to {exc.request.url.host.capitalize()} timed out"
+    else:
+        status_code = status.HTTP_502_BAD_GATEWAY
+        detail = f"{exc.request.url.host.capitalize()} service unexpectedly failed"
 
-    This handler is used as a "last resource" since it is recommended to handle these exceptions
-    closer to the raising point.
+    _logger.exception(
```

Comment:
This log should ONLY be for server side errors (i.e. 5XX). I see that `_handle_httpx_status_exceptions` can return som 4XX

Reply:
Good point. I corrected this.